linalg_ops/convolution/generate_e2e_conv2d_tests.py

```diff
@@ -0,0 +1,693 @@
+#!/usr/bin/env python3
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Generator for e2e conv2d tests.
+"""
+
+from typing import Optional
+import argparse
+import enum
+import dataclasses
+import typing
+import math
+import itertools
+import re
+
+# Data type of kernel entries. The string values must match MLIR data types.
+@enum.unique
+class KernelElemTypeId(enum.Enum):
+    NONE = ""
+    I8 = "i8"
+    F32 = "f32"
+    F16 = "f16"
+
+
+# Data type of input entries. The string values must match MLIR data types.
+@enum.unique
+class InputElemTypeId(enum.Enum):
+    NONE = ""
+    I8 = "i8"
+    F32 = "f32"
+    F16 = "f16"
```

Comment:
nit: I would order these by bit depth (8, 16, 32)

We could do the same in https://github.com/iree-org/iree-test-suites/blob/9f6bba6863f87faa788795e78e94676f7cc147c9/linalg_ops/matmul/generate_e2e_matmul_tests.py#L20-L27